src/fuzz/nxt_http_parse_fuzz.c

```diff
@@ -0,0 +1,106 @@
+
+/*
+ * Copyright (C) NGINX, Inc.
+ */
+
+#include <nxt_main.h>
+
+/* DO NOT TRY THIS AT HOME! */
+#include "nxt_controller.c"
+#include "nxt_h1proto.c"
+
+
+#define kMinInputLength 2
+#define kMaxInputLength 1024
+
+
+extern int LLVMFuzzerInitialize(int *argc, char ***argv);
+extern int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+
+extern char  **environ;
+
+
+int
+LLVMFuzzerInitialize(int *argc, char ***argv)
+{
+    nxt_lib_start("fuzzing", NULL, &environ);
+
+    if (nxt_slow_path(nxt_http_fields_hash(&nxt_controller_fields_hash,
+                               nxt_controller_request_fields,
+                               nxt_nitems(nxt_controller_request_fields)) != NXT_OK)) {
```

Comment:
Hmm, that's a little unwieldy.

I think it would be better to do the call to `nxt_http_fields_hash()` separately first then check it's retunr value.